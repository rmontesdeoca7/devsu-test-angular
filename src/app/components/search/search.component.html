<div class="search">
  <input
    type="text" 
    placeholder="Search..."
    (keyup)="search( txtInput.value )"
    #txtInput
    >
</div>
