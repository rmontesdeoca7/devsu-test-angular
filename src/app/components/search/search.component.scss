.search {
  width: 220px;
}