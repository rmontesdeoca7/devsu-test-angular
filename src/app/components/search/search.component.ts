import { Component, inject } from '@angular/core';
import { ApiService } from '../../services/api.service';

@Component({
  selector: 'app-search',
  templateUrl: './search.component.html',
  styleUrls: ['./search.component.scss']
})
export class SearchComponent {
  private apiService = inject(ApiService);

  search( term: string ) {
    this.apiService.dispachSearchTerm.emit( term.toLowerCase() );
  }
}
