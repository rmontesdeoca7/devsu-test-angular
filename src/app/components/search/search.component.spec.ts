import { ComponentFixture, TestBed } from '@angular/core/testing';

import { SearchComponent } from './search.component';
import { ApiService } from '../../services/api.service';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';

describe('SearchComponent', () => {
  let component: SearchComponent;
  let fixture: ComponentFixture<SearchComponent>;
  let compiled: HTMLElement;
  let service: ApiService;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [SearchComponent],
      imports: [
        HttpClientTestingModule
      ],
      providers: [
        ApiService
      ]
    });
    fixture = TestBed.createComponent(SearchComponent);
    component = fixture.componentInstance;
    service = TestBed.inject( ApiService );

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should be match with the snapshop', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  it('should emit search value to the service', () => {
    const value = 'txt';
    const spy = jest.spyOn(service.dispachSearchTerm, 'emit');
    component.search(value);
    expect(spy).toHaveBeenCalledWith(value);
  });
});
