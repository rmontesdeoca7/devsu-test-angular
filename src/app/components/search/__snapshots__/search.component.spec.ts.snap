// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent should be match with the snapshop 1`] = `"<div class="search"><input type="text" placeholder="Search..."></div>"`;
